packages/core/src/controllers/SwapController.ts

```diff
@@ -443,21 +446,40 @@ export const SwapController = {
   },
 
   async getInitialGasPrice() {
-    const res = await SwapApiUtil.fetchGasPrice()
+    const res = await SwapApiUtil.fetchGasPrice().catch(() => null)
 
     if (!res) {
-      return { gasPrice: null, gasPriceInUsd: null }
+      return { gasPrice: null, gasPriceInUSD: null }
     }
 
-    const value = res.standard
-    const gasFee = BigInt(value)
-    const gasLimit = BigInt(INITIAL_GAS_LIMIT)
-    const gasPrice = SwapCalculationUtil.getGasPriceInUSD(state.networkPrice, gasLimit, gasFee)
+    switch (NetworkController.state.caipNetwork?.chain) {
+      case 'solana':
+        state.gasFee = res.standard
+        state.gasPriceInUSD = NumberUtil.multiply(res.standard, state.networkPrice)
+          .dividedBy(1e9)
+          .toNumber()
 
-    state.gasFee = value
-    state.gasPriceInUSD = gasPrice
+        return {
+          gasPrice: BigInt(state.gasFee),
+          gasPriceInUSD: Number(state.gasPriceInUSD)
+        }
 
-    return { gasPrice: gasFee, gasPriceInUSD: state.gasPriceInUSD }
+      case 'evm':
```

Comment:
II think this is case is not needed.

Also, wouldn't it be better to split this into getSolanaGasFees and getEvmGasFees? Just to be tidy

Reply:
the `case 'evm'` is not needed, but it makes explicit that `evm` and default is the same